src/finch/compiled.py

```diff
@@ -29,6 +34,45 @@ def _to_lazy_tensor(x: Tensor | Any, /) -> Tensor | Any:
     return x if not isinstance(x, Tensor) else lazy(x)
 
 
+def _recurse_iter_return(x: IterObj, /, *, compute_kwargs: dict[str, Any]) -> IterObj:
```

Comment:
I would rather rename it to `_recurse_iter_compute` because we're computing here actual tensors. I don't think that `return` is informative in the function's name.

Reply:
Done in https://github.com/finch-tensor/finch-tensor-python/pull/94/commits/61455d58946d2f5fe8858c39eb9e14b55b941adc.